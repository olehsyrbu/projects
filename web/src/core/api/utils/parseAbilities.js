export function parseAbilities(abilities = []) {
  return abilities.map((ability) => {
    if (ability.conditions) {
      let array = ability.conditions.map((condition) => JSON.parse(condition));
      ability.conditions = Object.assign({}, ...array);
    }

    return {
      ...ability,
    };
  });
}
