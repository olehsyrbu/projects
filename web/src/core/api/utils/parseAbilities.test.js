import { parseAbilities } from './parseAbilities';

it('parse conditions to objects', () => {
  const rules = [
    {
      subject: 'Provider',
      action: ['read', 'unaffiliate', 'delete', 'paginate'],
      conditions: ['{"owner.organization.id":"571c9349-502a-47ce-9aee-24f780891fa8"}'],
    },
    {
      subject: 'Invitation',
      action: ['read', 'write', 'delete', 'create', 'paginate'],
      conditions: ['{"owner.organization.id":"571c9349-502a-47ce-9aee-24f780891fa8"}'],
    },
  ];

  const actual = parseAbilities(rules);

  expect(actual).toMatchSnapshot();
});

it('parse attributes without conditions', () => {
  const rules = [
    {
      subject: 'Provider',
      action: ['read', 'unaffiliate', 'delete', 'paginate'],
    },
    {
      subject: 'Invitation',
      action: ['read', 'write', 'delete', 'create', 'paginate'],
    },
  ];

  const actual = parseAbilities(rules);

  expect(actual).toMatchSnapshot();
});
