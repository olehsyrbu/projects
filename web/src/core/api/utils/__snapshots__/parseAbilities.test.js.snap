// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parse attributes without conditions 1`] = `
[
  {
    "action": [
      "read",
      "unaffiliate",
      "delete",
      "paginate",
    ],
    "subject": "Provider",
  },
  {
    "action": [
      "read",
      "write",
      "delete",
      "create",
      "paginate",
    ],
    "subject": "Invitation",
  },
]
`;

exports[`parse conditions to objects 1`] = `
[
  {
    "action": [
      "read",
      "unaffiliate",
      "delete",
      "paginate",
    ],
    "conditions": {
      "owner.organization.id": "571c9349-502a-47ce-9aee-24f780891fa8",
    },
    "subject": "Provider",
  },
  {
    "action": [
      "read",
      "write",
      "delete",
      "create",
      "paginate",
    ],
    "conditions": {
      "owner.organization.id": "571c9349-502a-47ce-9aee-24f780891fa8",
    },
    "subject": "Invitation",
  },
]
`;
